apps/transport/lib/transport/dataset.ex

```diff
@@ -313,7 +314,45 @@ defmodule Transport.Dataset do
     |> if do {:error, "Unable to validate dataset #{id}"} else {:ok, nil} end
   end
 
-  ## Private functions
+  def history_resources(%__MODULE__{} = dataset) do
+    if Application.get_env(:ex_aws, :access_key_id) == nil
+      || Application.get_env(:ex_aws, :secret_access_key) == nil do
+      # if the cellar credential are missing, we skip the whole history
+      []
+    else
+      try do
+        bucket = "dataset_#{dataset.datagouv_id}"
+        bucket
+        |> S3.list_objects
+        |> ExAws.stream!
+        |> Enum.to_list
+        |> Enum.map(fn f -> %{
+          name: f.key,
+          creation_date: f.last_modified,
+          href: history_resource_path(bucket, f.key),
+          metadata: fetch_metadata(bucket, f.key),
+          } end)
+        rescue
+          e in ExAws.Error -> Logger.error("error while accessing the S3 bucket: #{inspect e}")
+          []
+        end
+      end
+    end
+
+    ## Private functions
+    @cellar_host ".cellar-c2.services.clever-cloud.com/"
+
+  defp history_resource_path(bucket, name), do: Path.join(["http://", bucket, @cellar_host, name])
+
+  defp fetch_metadata(bucket, obj_key) do
+    bucket
+      |> S3.head_object(obj_key)
+      |> ExAws.request!
+      |> Map.get(:headers)
+      |> Enum.into(%{}, fn {k, v} -> {String.replace(k, "x-amz-meta-", ""), v} end)
```

Comment:
it's the same, but you could have use [Map.new/2](https://hexdocs.pm/elixir/Map.html#new/2) instead